### PR TITLE
winbuild: fail to build WITH_ZLIB=static

### PR DESCRIPTION
**Pull request by moehuster**

### CODE REVIEW

File: winbuild/MakefileBuild.vc [hunk 1]
```diff
@@ -262,8 +262,8 @@ ZLIB_LIBS   = zlib.lib
 USE_ZLIB    = true
 ZLIB        = dll
 !ELSEIF "$(WITH_ZLIB)"=="static"
-!IF EXISTS("$(ZLIB_LIB_DIR)\zlibstat.lib")
-ZLIB_LIBS   = zlibstat.lib
```
Comment (jay) [inline winbuild/MakefileBuild.vc:266]: Not a typo, the visual studio zlibstat project's output is [zlibstat.lib](https://github.com/madler/zlib/blob/04f42ceca40f73e2978b50e93806c2a18c1281fc/contrib/vstudio/vc10/zlibstat.vcxproj#L182). Where are you getting zlibstatic.lib from? Are you using a different build method?

**Comment by moehuster** [comment]
I use cmake generate visual studio project. It's zlib's bug, may be.
`add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_PUBLIC_HDRS} ${ZLIB_PRIVATE_HDRS})`
[https://github.com/madler/zlib/blob/master/CMakeLists.txt#L151](url)

File: winbuild/MakefileBuild.vc [hunk 2]
```diff
@@ -265,6 +265,8 @@ ZLIB        = dll
 !IF EXISTS("$(ZLIB_LIB_DIR)\zlibstat.lib")
 ZLIB_LIBS   = zlibstat.lib
 ADDITIONAL_ZLIB_CFLAGS = /DZLIB_WINAPI
+!ELSEIF EXISTS("$(ZLIB_LIB_DIR)\zlibstatic.lib")
+ZLIB_LIBS   = zlibstatic.lib
```
[the remark this answers is not in the document]
Comment (MarcelRaad) [reply winbuild/MakefileBuild.vc:269]: Yes, I can confirm that `ZLIB_WINAPI` is only defined by the VS projects by default.

**Comment by jay** [comment]
Thanks

**Pull request closed without merging**